extendr-macros/src/wrappers.rs

```diff
@@ -354,7 +354,17 @@ pub fn translate_formal(input: &FnArg, self_ty: Option<&syn::Type>) -> syn::Resu
         // function argument.
         FnArg::Typed(ref pattype) => {
             let pat = &pattype.pat.as_ref();
-            Ok(parse_quote! { #pat : extendr_api::SEXP })
+            // ensure that `mut` in args are ignored in the wrapper
+            let pat_ident = match pat {
+                syn::Pat::Ident(ref pat_ident) => &pat_ident.ident,
+                _ => {
+                    return Err(syn::Error::new_spanned(
+                        input,
+                        "failed to translate name of argument",
+                    ));
+                }
+            };
```

Comment:
Duplicate code fragment

Reply:
What do you suggest I do?

Reply:
Ideally of course extract it as a function, but if it cannot be done (I do not see exact types here so maybe there is a mismatch), then let's leave it as is

Reply:
Good suggestion; I personally don't like how many functions this part of extendr has.. but I think I've made it work! Please check the latest commit.